core/domain/feedback_domain.py

```diff
@@ -362,6 +363,36 @@ def to_dict(self) -> FeedbackMessageReferenceDict:
             'message_id': self.message_id
         }
 
+    @classmethod
+    def from_dict(
+        cls,
+        reference_dict: FeedbackMessageReferenceDict
+        ) -> FeedbackMessageReference:
+        """Return a FeedbackMessageReference domain object from a dict.
+
+        Args:
+            reference_dict: dict. The dict representation of a
+                FeedbackMessageReference object.
+
+        Returns:
+            FeedbackMessageReference. The corresponding FeedbackMessageReference
+            domain object.
+        """
+        # Investigate why reference_dict is passed as
+        # string and not FeedbackMessageReferenceDict when called from Postman.
+        if isinstance(reference_dict, str):
```

Comment:
Somehow, when using Postman to create a POST request to this URL, the `reference_dict` passed to this method is of type `string` and not `FeedbackMessageReference`. Is this intentional?

Reply:
Not sure can you please share a screenshot?

Reply:
Here's the POST request:

<img width="912" alt="Skärmavbild 2022-11-11 kl  08 37 45" src="https://user-images.githubusercontent.com/48027815/201290287-f8b12c89-22f0-4625-8750-df43441771ff.png">

And this is where the `reference_dict` (obj) is passed to the `from_dict`: in `schema_utils.py`

<img width="688" alt="Skärmavbild 2022-11-11 kl  08 40 11" src="https://user-images.githubusercontent.com/48027815/201290703-f71db40d-f223-455e-9cbf-8852ff05ea20.png">

Reply:
@Nik-09, Any suggestions?

Also, now when `from_dict` and `validate` has been called successfully on the FeedbackMessageReference object the response from the POST request is 
```json
{
"error": "You do not have the credentials to access this page.",
"status_code": 401
}
```

Reply:
Yes, got the point. The user ID that you have entered does not have the right to access the page. 

You can start the dev server and assign rights from the admin roles tab. Then from the UI try to hit the handler.
Hope this helps.

Feel free to ask if you have any questions.
Thank you

Reply:
Okay, will try that. Thank you!

How about the original issue: the `from_dict` method is passed the dict as a string, and not as a FeedbackMessageReferenceDict. That's why I had to add the conversion. Shouldn't the reference_dict be of the correct type before calling `from_dict`  in `schema_utils.py`?

Reply:
Or is it okay to do the conversion?

Reply:
The reference_dict should be of type dict, not string. 
As you will be accessing some of the fields from the dict in the handler class.

The schema validation will normalize the dict into a `FeedbackMessageReference` object as per the provided schema. 

I am a bit unfamiliar with postman so not able to guide you in this respect but in general, try to reverify the parameters and methods before making an HTTP request.

Also, I think this problem is taking longer than expected, we can hop on a short call if required.

Thank you